### PR TITLE
Update docs about EQL CCS

### PR DESCRIPTION
**Pull request by luigidellaquila**

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/rest-api-changes.asciidoc [hunk 1]
```diff
@@ -196,8 +196,8 @@ remote 8.x cluster.
 
 include::{es-repo-dir}/search/search-your-data/ccs-version-compat-matrix.asciidoc[]
 
-IMPORTANT: For the {ref}/eql-search-api.html[EQL search API], the local and
-remote clusters must use the same {es} version.
+IMPORTANT: For the {ref}/eql-search-api.html[EQL search API] specific limitations apply,
+see {ref}/eql.html#run-eql-search-across-clusters[Run an EQL search across clusters]
```
Comment (astefan) [inline docs/reference/migration/migrate_8_0/rest-api-changes.asciidoc:200]:
```suggestion
IMPORTANT: For the {ref}/eql-search-api.html[EQL search API], the local and
remote clusters must use the same {es} version if they have versions prior to 7.17.7 (included) or prior to 8.5.1 (included).
```

File: docs/reference/search/search-your-data/search-across-clusters.asciidoc [hunk 2]
```diff
@@ -445,8 +445,8 @@ remote 8.x cluster.
 [[ccs-version-compatibility]]
 include::{es-repo-dir}/search/search-your-data/ccs-version-compat-matrix.asciidoc[]
 
-IMPORTANT: For the <<eql-search-api,EQL search API>>, the local and remote
-clusters must use the same {es} version.
+IMPORTANT: For the <<eql-search-api,EQL search API>>,
+specific limitations apply, see <<run-eql-search-across-clusters>>.
```
Comment (astefan) [inline docs/reference/search/search-your-data/search-across-clusters.asciidoc:449]:
```suggestion
IMPORTANT: For the <<eql-search-api,EQL search API>>, the local and remote
clusters must use the same {es} version if they have versions prior to 7.17.7 (included) or prior to 8.5.1 (included).
```

**Review by astefan — COMMENTED** [review]
I've left a proposal regarding the wording:
- instead of pointing to the ccs versions matrix, I've kept the same wording as before with an exception regarding the two versions it applies
- I removed the redirection to one message only. This combined with a simpler message regarding CCS versions, I'm hoping it aims for a simpler/clearer user experience when consuming the documentation.

**Comment by luigidellaquila** [comment]
Thanks @astefan, it makes the message more readable and straight-forward. 
I committed you proposed changes as they are

File: docs/reference/eql/eql.asciidoc [hunk 3]
```diff
@@ -797,9 +797,12 @@ You can also manually delete saved synchronous searches using the
...
 
-The EQL search API supports <<modules-cross-cluster-search,cross-cluster
-search>>. However, the local and <<remote-clusters,remote clusters>>
-must use the same {es} version.
+The EQL search API supports <<modules-cross-cluster-search,cross-cluster search>>. However, the local and <<remote-clusters,remote clusters>> must use the same {es} version if they have versions prior to 7.17.7 (included) or prior to 8.5.1 (included).
+search>> with <<remote-clusters,remote clusters>> v 7.17.8+,
+and limited to <<ccs-supported-configurations>>.
+However, if local node has version prior to 7.17.8 or 8.5.2, <<remote-clusters,remote clusters>>
+must be the same {es} version to run <<modules-cross-cluster-search,cross-cluster
+search>>.
```
Comment (astefan) [inline docs/reference/eql/eql.asciidoc:805]: This one is not right. It got messed up when the suggestion was applied.

Comment (luigidellaquila) [reply docs/reference/eql/eql.asciidoc:805]: fixing

**Review by astefan — APPROVED** [review]
LGTM

**Review by bpintea — APPROVED** [review]
LGTM

**Pull request merged**